### PR TITLE
recipes/70-test_ssl{cbcpadding,extension,records}: make it work w/fragmentation.

### PR DESCRIPTION
**Pull request by dot-asm**
This fixes only those tests that were failing when network data was
fragmented. Remaining ones might succeed for "wrong reasons". Bunch
of tests have to fail to be considered successful and when data is
fragmented they might fail for reasons other than originally intended.

make test TESTS=70 works even with 1-byte fragmentation...

##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/recipes/70-test_sslcbcpadding.t [hunk 1]
```diff
@@ -41,26 +41,32 @@ my @test_offsets = (0, 128, 254, 255);
...
-    $proxy->start();
-    ok(TLSProxy::Message->fail(), "Invalid padding byte $bad_padding_offset");
+    $fatal_alert = 0;
+    $proxy->clearClient();
+    $proxy->clientstart();
+    ok($fatal_alert, "Invalid padding byte $bad_padding_offset");
 }
 
+{
+my $sent_corrupted_payload;
```
Comment (levitte) [inline test/recipes/70-test_sslcbcpadding.t]: Use the [state](https://perldoc.perl.org/functions/state.html) feature instead.

File: test/recipes/70-test_sslextension.t [hunk 2]
```diff
@@ -98,38 +99,58 @@ sub inject_duplicate_extension_clienthello
...
         return;
     }
 
-    inject_duplicate_extension($proxy, TLSProxy::Message::MT_SERVER_HELLO);
+    my $last_record = @{$proxy->{record_list}}[-1];
+    $fatal_alert = 1 if $last_record->is_fatal_alert(0);
 }
 
+{
+my $sent_unsolisited_extension;
```
Comment (levitte) [inline test/recipes/70-test_sslextension.t]: Use the [state](https://perldoc.perl.org/functions/state.html) feature instead.

File: test/recipes/70-test_sslrecords.t [hunk 3]
```diff
@@ -427,20 +441,26 @@ sub add_sslv2_filter
 
 }
 
+{
+my $added_record;
```
Comment (levitte) [inline test/recipes/70-test_sslrecords.t]: Use the [state](https://perldoc.perl.org/functions/state.html) feature instead.

**Comment by bernd-edlinger** [comment]
Have you actually ever observed such short fragments?
On windows the local connection uses an MTU of 1460 bytes.
Linux transfers even >64K in one chunk.

**Comment by dot-asm** [comment]
> Have you actually ever observed such short fragments?

Of course not. It's just [artificial] 1-byte fragmentation is meaningful thing to test. Because it works as insurance that there are no unexpected surprises when fragmentation happen to coincide with record boundaries.

On related note, to question of 0e3ecaec3ebf2ed0dfbcff07b7fa1cfdd8240d5c vs. c53c2fec82b6026331c98642ce4ad32ff7fe4fef. 1-byte fragmentation makes data flow equivalent to the latter, i.e. forwarding complete records as they become available...

File: test/recipes/70-test_sslcbcpadding.t (hunk 1 above)
Comment (dot-asm) [reply test/recipes/70-test_sslcbcpadding.t]: Like that?

**Pull request closed without merging**